src/shared_modules/utils/socketWrapper.hpp

```diff
@@ -110,6 +110,163 @@ class TcpAddress final : public SockAddress<TcpAddress>
     }
 };
 
+/**
+ * @brief This class is used for the following format:
+ *          - 4 bytes: packet size. Mandatory size of the whole packet (body, optional header and header size)
+ *          - 4 bytes: header size. Mandatory size of the optional header. It can be 0.
+ *          - N bytes: header. Optional header.
+ *          - M bytes: body. Mandatory data to send.
+ */
+class appendHeaderProtocol final
+{
+public:
+    /**
+     * @brief Create the buffer to send according to this protocol.
+     *
+     * @param buffer        Output buffer to write.
+     * @param bufferSize    Size of the output buffer.
+     * @param dataBody      Data to send.
+     * @param sizeBody      Size of the data to send.
+     * @param dataHeader    Optional header to send.
+     * @param sizeHeader    Size of the optional header.
+     */
+    void static buildBuffer(std::vector<char>& buffer,
+                            uint32_t& bufferSize,
+                            const char* dataBody,
+                            uint32_t sizeBody,
+                            const char* dataHeader = nullptr,
+                            uint32_t sizeHeader = 0)
+    {
+        if (PACKET_SIZE + HEADER_SIZE + sizeHeader + sizeBody > BUFFER_SIZE)
```

Comment:
I would suggest more precise names for the following constants:
```
PACKET_SIZE -> PACKET_FIELD_SIZE
HEADER_SIZE -> HEADER_FIELD_SIZE
BUFFER_SIZE -> BUFFER_MAX_SIZE
```

Reply:
It's a good idea, it was hard to understand it the first time I saw it.